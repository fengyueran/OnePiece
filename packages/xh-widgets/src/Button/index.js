import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import ButtonBase from '../ButtonBase';
import { fade } from '../utils/colorManipulator';

const ButtonStyle = css`
  min-width: 64px;
  min-height: 36px;
  padding: 8px 16px;
  border-radius: 2px;
  margin: 8px;
  font-size: 0.875rem;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-weight: 500;
  line-height: 1.5;
  border-radius: 4px;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  background-color: #e0e0e0; 
  color: rgba(0, 0, 0, 0.87);
  box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12);
  &:hover {
    background-color: ${fade('rgba(0, 0, 0, 0.87)', 0.2)};
  }
`;

const Container = styled.span`
  width: 100%;
  display: inherit;
  align-items: inherit;
  justify-content: inherit;
`;

const Button = ({ 
  isRipple, children, style, className, onClick
}) => (
  <ButtonBase
    onClick={onClick} 
    style={style}
    cssStyle={ButtonStyle}
    className={className}
    isRipple={isRipple}
  >
    <Container>{children}</Container>
  </ButtonBase>
);

Button.propTypes = {
  isRipple: PropTypes.bool,
  style: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.node,
  onClick: PropTypes.func,
};

export { Button };
export default Button;