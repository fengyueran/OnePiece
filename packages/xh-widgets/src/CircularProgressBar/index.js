import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Container = styled.div`
  display: 'inline-block';
  line-height: 1;
`;

const SIZE = 44;

const CircularProgressBar = props => {
  const { size, thickness, value, trailColor, pathColor } = props;
  const rootStyle = {};
  const circleStyle = {};

  rootStyle.transform = 'rotate(-90deg)';
  const circumference = 2 * Math.PI * ((SIZE - thickness) / 2);
  circleStyle.strokeDasharray = circumference.toFixed(3);
  circleStyle.strokeDashoffset = `${(
    ((100 - value) / 100) *
    circumference
  ).toFixed(3)}px`;

  return (
    <Container
      style={{ width: size, height: size, ...rootStyle }}
      role="progressbar"
    >
      <svg viewBox={`${SIZE / 2} ${SIZE / 2} ${SIZE} ${SIZE}`}>
        <circle
          cx={SIZE}
          cy={SIZE}
          r={(SIZE - thickness) / 2}
          fill="none"
          stroke={trailColor}
          strokeWidth={thickness}
        />
        <circle
          style={circleStyle}
          cx={SIZE}
          cy={SIZE}
          r={(SIZE - thickness) / 2}
          fill="none"
          stroke={pathColor}
          strokeWidth={thickness}
        />
      </svg>
    </Container>
  );
};

CircularProgressBar.propTypes = {
  size: PropTypes.number,
  thickness: PropTypes.number,
  value: PropTypes.number,
  pathColor: PropTypes.string,
  trailColor: PropTypes.string
};

CircularProgressBar.defaultProps = {
  size: 40,
  thickness: 3.6,
  value: 0,
  pathColor: '#51B5F4',
  trailColor: '#F5F5F5'
};

export default CircularProgressBar;
