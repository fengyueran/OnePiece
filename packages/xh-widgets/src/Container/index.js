export { 
  FlexContainer,
  VContainer,
  LineContainer
} from './Container';