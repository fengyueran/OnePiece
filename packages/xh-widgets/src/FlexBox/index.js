import styled from 'styled-components';

const FlexBox = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  width: 100%;
`;

export { FlexBox };
export default FlexBox;
