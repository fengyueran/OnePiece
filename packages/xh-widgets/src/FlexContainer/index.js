import styled from 'styled-components';

const FlexContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  width: 100%;
`;

export { FlexContainer };
export default FlexContainer;