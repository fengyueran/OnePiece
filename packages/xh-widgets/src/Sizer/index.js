import styled from 'styled-components';

const Sizer = styled.div`
  flex-grow: 1;
`;

Sizer.X = styled.div`
  width: ${props => `${props.size}px`};
`;

Sizer.Y = styled.div`
  height: ${props => `${props.size}px`};
`;

export default Sizer;
