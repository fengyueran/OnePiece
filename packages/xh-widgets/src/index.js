export * from './FlexContainer';
export * from './LineContainer';
export * from './VContainer';
export * from './CircularProgressBar';
export * from './Button';
export * from './utils';