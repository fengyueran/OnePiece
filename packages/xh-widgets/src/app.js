
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';
import { Button } from './Button';

const StyledButton = styled(Button)`
  background: #536DFE;
`;
class App extends Component {
  render() {
    return (
      <div style={{ padding: 20 }}>
        <StyledButton onClick={() => alert('clicked')}>
          Submit
        </StyledButton>
      </div>
    );
  }
}

ReactDOM.render(<App />, document.getElementById('root'));

export default App;
