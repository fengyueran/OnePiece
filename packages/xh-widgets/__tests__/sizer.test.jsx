import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Sizer, SizerX, SizerY } from '../src/Sizer';

const setup = (Component, props) => {
  const wrapper = shallow(
    <Component {...props} />
  );
  return wrapper;
};

describe('VerticalBox Shallow', () => {
  const sizerWrapper = setup(Sizer, { id: 'sizer' });
  it('Sizer should be render', () => {
    expect(sizerWrapper.find('#sizer').exists()).toBe(true); 
  });

  const sizerXWrapper = setup(SizerX, { id: 'sizerX', size: 18 });
  it('SizerX should be render', () => {
    expect(sizerXWrapper.find('#sizerX').props().size).toBe(18); 
  });

  const sizerYWrapper = setup(SizerY, { id: 'sizerY', size: 20 });
  it('SizerY should be render', () => {
    expect(sizerYWrapper.find('#sizerY').props().size).toBe(20); 
  });

  it('Sizer snapshot', () => {
    const Instance = (
      <div>
        <Sizer />
        <SizerX size={18} />
        <SizerY size={28} />
      </div>);
    const tree = renderer
      .create(Instance)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});