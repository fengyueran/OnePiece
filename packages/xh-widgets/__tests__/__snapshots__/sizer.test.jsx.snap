// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalBox Shallow Sizer snapshot 1`] = `
<div>
  <div
    className="sc-bdVaJa kUhAfA"
  />
  <div
    className="sc-bwzfXH hvNiVj"
    size={18}
  />
  <div
    className="sc-htpNat enWhWs"
    size={28}
  />
</div>
`;
